hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/table/AbstractTable.java

```diff
@@ -448,6 +496,25 @@ protected UINotification getNotification() {
         return notification;
     }
 
+    /**
+     * Finds the entity object of the given entity ID and performs the
+     * publishing of the BaseEntityEventType.SELECTED_ENTITY event
+     * 
+     * @param entityId
+     *            ID of the current entity
+     * @param tableClass
+     *            the class of the table where the entity was selected in. Is
+     *            necessary to publish the correct event, in case the entity
+     *            value is null --> clear the entity details
+     */
+    public void selectEntity(final I entityId, final Class<?> tableClass) {
```

Comment:
Sorry don't understand, why do you need the table Class as parameter? It's the same like `this.getClass()`?